### PR TITLE
refactor(module:select): migrate to new control flow

### PR DESCRIPTION
**Pull request by HyperLife1119**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Application (the showcase website) / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

**Review by Laffery — APPROVED** [review]
lgtm

**Comment by Laffery** [comment]
there are some ci tests failed, please fix them  first

File: components/select/select.spec.ts [hunk 1]
```diff
@@ -82,10 +84,10 @@ describe('select', () => {
       fixture.detectChanges();
       flush();
       fixture.detectChanges();
-      expect(selectElement.querySelector('nz-select-item')!.textContent).toBe('test_label');
+      expect(selectElement.querySelector('nz-select-item')!.textContent?.trim()).toBe('test_label');
```
Comment (HyperLife1119) [inline components/select/select.spec.ts:87]: 由于新的控制流语法会被格式化换行，导致 textContent 字符串前后各存在一个空白符，这里通过统一调用 trim() 来消除前后空格

**Review by Laffery — APPROVED** [review]
LGTM

**Pull request merged**